diff --git a/includes/common.inc b/includes/common.inc
index 477ecc0..9419cdd 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -7926,14 +7926,14 @@ function entity_uri($entity_type, $entity) {
     return NULL;
   }
 
-  // Invoke the callback to get the URI. If there is no callback, return NULL.
   if (isset($uri_callback) && function_exists($uri_callback)) {
-    $uri = $uri_callback($entity);
-    // Pass the entity data to url() so that alter functions do not need to
-    // lookup this entity again.
-    $uri['options']['entity_type'] = $entity_type;
-    $uri['options']['entity'] = $entity;
-    return $uri;
+    if ($uri = $uri_callback($entity)) {
+      // Pass the entity data to url() so that alter functions do not need to
+      // lookup this entity again.
+      $uri['options']['entity_type'] = $entity_type;
+      $uri['options']['entity'] = $entity;
+      return $uri;
+    }
   }
 }
 
diff --git a/modules/rdf/rdf.module b/modules/rdf/rdf.module
index 877b598..b426621 100644
--- a/modules/rdf/rdf.module
+++ b/modules/rdf/rdf.module
@@ -576,36 +576,37 @@ function rdf_preprocess_field(&$variables) {
  */
 function rdf_preprocess_user_profile(&$variables) {
   $account = $variables['elements']['#account'];
-  $uri = entity_uri('user', $account);
 
   // Adds RDFa markup to the user profile page. Fields displayed in this page
   // will automatically describe the user.
   if (!empty($account->rdf_mapping['rdftype'])) {
     $variables['attributes_array']['typeof'] = $account->rdf_mapping['rdftype'];
+  }
+  if ($uri = entity_uri('user', $account)) {
     $variables['attributes_array']['about'] = url($uri['path'], $uri['options']);
+    // Adds the relationship between the sioc:UserAccount and the foaf:Person who
+    // holds the account.
+    $account_holder_meta = array(
+      '#tag' => 'meta',
+      '#attributes' => array(
+        'about' => url($uri['path'], array_merge($uri['options'], array('fragment' => 'me'))),
+        'typeof' => array('foaf:Person'),
+        'rel' => array('foaf:account'),
+        'resource' => url($uri['path'], $uri['options']),
+      ),
+    );
+    // Adds the markup for username.
+    $username_meta = array(
+      '#tag' => 'meta',
+      '#attributes' => array(
+        'about' => url($uri['path'], $uri['options']),
+        'property' => $account->rdf_mapping['name']['predicates'],
+        'content' => $account->name,
+      )
+    );
+    drupal_add_html_head($account_holder_meta, 'rdf_user_account_holder');
+    drupal_add_html_head($username_meta, 'rdf_user_username');
   }
-  // Adds the relationship between the sioc:UserAccount and the foaf:Person who
-  // holds the account.
-  $account_holder_meta = array(
-    '#tag' => 'meta',
-    '#attributes' => array(
-      'about' => url($uri['path'], array_merge($uri['options'], array('fragment' => 'me'))),
-      'typeof' => array('foaf:Person'),
-      'rel' => array('foaf:account'),
-      'resource' => url($uri['path'], $uri['options']),
-    ),
-  );
-  // Adds the markup for username.
-  $username_meta = array(
-    '#tag' => 'meta',
-    '#attributes' => array(
-      'about' => url($uri['path'], $uri['options']),
-      'property' => $account->rdf_mapping['name']['predicates'],
-      'content' => $account->name,
-    )
-  );
-  drupal_add_html_head($account_holder_meta, 'rdf_user_account_holder');
-  drupal_add_html_head($username_meta, 'rdf_user_username');
 }
 
 /**
